pkg/gator/test/types.go

```diff
@@ -10,7 +10,14 @@ import (
 type GatorResult struct {
 	types.Result
 
-	ViolatingObject *unstructured.Unstructured
+	ViolatingObject *unstructured.Unstructured `json:"violatingObject"`
+
+	// Trace is an explanation of the underlying constraint evaluation.
```

Comment:
I wish this `Trace` and the `CaseResult.Trace` comments where not copy pasted but I couldn't think of a way to do that without a wholesale refactor.

Let me know y'all if you see a quick, small change that could remove the duplication or if it's acceptable in this case.

Reply:
IMO copy/paste is fine for this.

If this were for a K8s resource and the comments were used for doc gen... maybe an embeded struct?

```
type Dedup struct {
   // Name is super meaningful
   Name string
}


type A struct {
   Dedup
   
   Something string
}

type B struct {
   Dedup
   
   DifferentThing string
}
```

But that seems a bit heavy weight here.

Reply:
I can give this a shot but I think (haven't tried it yet) that the indirection will not play nicely w the yaml processing (just how we had `violatingObject: object: ...`; but let's try it out

Reply:
So I had this setup 

```
type TraceWrapper struct {
  // comment we want not copied
  Trace *string
}
```

and embedded that into `GatorResult` as 

```
type GatorResult struct {
  // things that where hear before
  TraceWrapper 
}
```

Unfortunately, without specifying the marshaling rules, I think, the JSON will look like `trace: {trace: {}}`

---

That said, we could do the same trick as "YamlGatorResult" and call it "MarshalFiendlyGatorResult" or some other better name where we remove that level of indirection like we do for `violatingObject` right now.

Wdyt?

Reply:
FWIW I'd just copy/paste for this (embedding structs in other structs has its own readability/usability penalty), but was giving you an example of how deduplication could be done since you were asking. K8s uses this for metadata:


https://github.com/kubernetes/apimachinery/blob/b03a432a2a6dc35338232d6aa849e18d29e72ce2/pkg/apis/meta/v1/types.go#L1038-L1059

Where they use this json tag to control serialization `TypeMeta `json:",inline"``

TL;DR, just copy/paste the comment, much less effort and not much is harmed if they drift.